openfisca_aotearoa/variables/acts/social_security/supported_living_payment/supported_living_payment.py

```diff
@@ -1,124 +1,236 @@
     value_type = bool
-    default_value = False
     entity = entities.Person
-    label = "Person is currently granted the supported living benefit"
     definition_period = periods.WEEK
-    reference = "Reference is unclear, but variable is utilised by the phrase: 'granted a main benefit'"
+    set_input = holders.set_input_dispatch_by_period
+    label = "Eligible for Supported Living Payment."
+    reference = [
+        "https://legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783175",
+        "http://legislation.govt.nz/act/public/1964/0136/latest/whole.html#DLM5468367",
+        ]
 
+    def formula_2013_07_15(persons, period, parameters):
+        """
+        For the 1964 version of the act (final version), this is split into two sections:
+            40B on ground of sickness, injury, disability, or total blindness
+            40D on ground of caring for patient requiring care
+        """
+        ssa40B = persons("supported_living_payment__disabled_or_blind__entitled", period, parameters)
+        ssa40D = persons("supported_living_payment__carer__entitled", period, parameters)
+        return ssa40B + ssa40D
 
-class supported_living_payment__receiving(variables.Variable):
+    def formula_2018_11_26(persons, period, parameters):
+        """
+        For the 2018 version of the act, this is split into two sections:
+            34 - 39 on ground of restricted work capacity or total blindness
+            40 - 42 on ground of caring for another person
+        """
+        ssa34 = persons("supported_living_payment__disabled_or_blind__entitled", period, parameters)
+        ssa40 = persons("supported_living_payment__carer__entitled", period, parameters)
+
+        # ssa116 - must undergo work ability assessment
+
+        # ssa117 - unless they are terminally ill or their condition will not "improve"
+
+        # ssa121_e - spouse must undertake work-preparation obligations
+
+        # ssa128 - children must atend school & healthcare, person required to attend interviews
+
+        # ssa140 - spouse must be available to work
+
+        # ssa170 - person or spouse must work with service providers
+
+        return ssa34 + ssa40
+
+
+class supported_living_payment__disabled_or_blind__entitled(variables.Variable):
     value_type = bool
-    default_value = False
     entity = entities.Person
-    label = "Person is currently recieving/being paid the supported living payment"
     definition_period = periods.WEEK
-    reference = "Reference is unclear, but concept underpinning the variable assumes it covers both: 'being paid a main benefit' or 'recieving a benefit'"
+    label = "Eligible for Supported Living Payment, on ground of restricted work capacity or total blindness."
+    reference = [
+        "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783174",
+        ]
 
+    def formula_2018_11_26(persons, period, parameters):
```

Comment:
Just want to say how great it is to see the internal logic laid out like this, makes reviewing much easier!